corsikaio/subblocks/data.py

```diff
@@ -1,33 +1,39 @@
-import numpy as np
+from .dtypes import build_dtype, Field
 
-cherenkov_photons_dtype = np.dtype([
-    ('n_photons', 'float32'),
-    ('x', 'float32'),
-    ('y', 'float32'),
-    ('u', 'float32'),
-    ('v', 'float32'),
-    ('t', 'float32'),
-    ('production_height', 'float32'),
-])
+cherenkov_photons_fields = [
+    Field(1, "n_photons", dtype="float32"),
+    Field(2, "x", unit="cm", dtype="float32"),
+    Field(3, "y", unit="cm", dtype="float32"),
+    Field(4, "u", dtype="float32"),
+    Field(5, "v", dtype="float32"),
+    Field(6, "t", unit="ns", dtype="float32"),
+    Field(7, "production_height", unit="cm", dtype="float32"),
+]
 
-mmcs_cherenkov_photons_dtype = np.dtype([
-    ('n_photons', 'float32'),
-    ('x', 'float32'),
-    ('y', 'float32'),
-    ('u', 'float32'),
-    ('v', 'float32'),
-    ('t', 'float32'),
-    ('wavelength', 'float32'),
-    ('mother_particle', 'uint16'),
-    ('production_height', 'float32'),
-])
+cherenkov_photons_dtype = build_dtype(cherenkov_photons_fields)
```

Comment:
You need to pass itemsize=None here, the default is 273 * 4, which is the whole block.

Reply:
Fixed